macros/parserFormulaWithUnits.pl

```diff
@@ -46,6 +75,15 @@ =head1 USAGE
  #
 
 sub _parserFormulaWithUnits_init {
+    # We make copies of these hashes here because these copies will be unique to  # the problem.  The hashes in Units are shared between problems.  We pass
+  # the hashes for these local copies to the NumberWithUnits package to use
+  # for all of its stuff.  
+  %fundamental_units = %Units::fundamental_units;
+  %known_units = %Units::known_units;
```

Comment:
It looks like these are setting global values in the `main::` namespace.  Should these be `my` variables?